internal/controllers/helpers.go

```diff
@@ -0,0 +1,235 @@
+/*
```

Comment:
I think that having helpers here is unnessesary. All of these methods were previously a part of reconcile object, they still are. It may be we only need to have a selector on the v1 or v3 cluster object inside the structure, to decide logic path.

Reply:
This was just an effort to reduce code duplication. I'd say that, if we're deprecating the `provisioning.cattle.io` controller in the near future, we can just address this formatting when we do that. Then we can use a structure equivalent to what we've had until now.